python/pyspark/sql/tests/connect/test_connect_basic.py

```diff
@@ -302,6 +301,31 @@ def test_to_pandas(self):
             self.spark.sql(query).toPandas(),
         )
 
+    def test_select_expr(self):
+        # SPARK-41201: test selectExpr API.
+        self.assert_eq(
+            self.connect.read.table(self.tbl_name).selectExpr("id * 2").toPandas(),
+            self.spark.read.table(self.tbl_name).selectExpr("id * 2").toPandas(),
+        )
+        self.assert_eq(
+            self.connect.read.table(self.tbl_name)
+            .selectExpr(["id * 2", "cast(name as long) as name"])
+            .toPandas(),
+            self.spark.read.table(self.tbl_name)
+            .selectExpr(["id * 2", "cast(name as long) as name"])
+            .toPandas(),
+        )
+
+        self.assert_eq(
+            self.connect.read.table(self.tbl_name)
+            .selectExpr("id * 2", "cast(name as long) as name")
+            .toPandas(),
+            self.spark.read.table(self.tbl_name)
+            .selectExpr("id * 2", "cast(name as long) as name")
+            .toPandas(),
+        )
+
+    @unittest.skip("test_fill_na is flaky")
```

Comment:
@amaliujia  why you disable this test again?

Reply:
ah .. I didn't notice this. Can we enable this back?

Reply:
I think so, will send a followup for it

Reply:
I am pretty sure I removed this after conflict resolution. 

Actually Martin pointed out another case: https://github.com/apache/spark/pull/38723#discussion_r1028942189

Basically it seems happened more than once that after code conflict resolution, the code I want to keep is gone.|

Maybe I should always do a `-i` commits square to in case more than 1 commit rebase causing unexpected result.

Reply:
I will follow up this soon.

Reply:
I am really guessing if I have more than 1 commit locally, if the first one I resolve the conflict, the following commit that might add something back silently.....

Reply:
no worries, add it back in https://github.com/apache/spark/pull/38763